### ISSUE TITLE
DL3018 is reported when locking apk packageversion by `~` (tilde)

### ISSUE THREAD

**Issue by yukihiko-shinoda** [hadolint/hadolint#1165]
Acording the histories of #295 , we should use `~` to lock apk package version:

- https://github.com/hadolint/hadolint/issues/295#issue-418483291
- https://github.com/hadolint/hadolint/issues/295#issuecomment-470898217
- https://github.com/hadolint/hadolint/issues/295#issuecomment-471012358
- https://github.com/hadolint/hadolint/issues/295#issuecomment-472097553

In spite of that, Hadolint reports [DL3018](https://github.com/hadolint/hadolint/wiki/DL3018):
[DL3018 · hadolint/hadolint Wiki](https://github.com/hadolint/hadolint/wiki/DL3018)

I use Hadolint 2.14.0:

```console
$ hadolint --version
Haskell Dockerfile Linter 2.14.0
```

Step to reproduce:

Dockerfile:

```docker
FROM alpine:3.23.2
RUN apk add --no-cache git~2.52.0
```

Then:

```
$ hadolint Dockerfile.test                                                                
Dockerfile.test:2 DL3018 warning: Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```

**Comment by nothankyouzzz** [comment]
I opened a PR to fix this issue: #1168 .

It updates DL3018 to treat `apk add pkg~version` as pinned and adds a test case.

Feedback welcome!
